packages/cactus-common/src/main/typescript/logging/logger.ts

```diff
@@ -21,6 +22,61 @@ export interface ILoggerOptions {
   level?: LogLevelDesc;
 }
 
+/**
+ * STANDARD EXCEPTION HANDLING - EXAMPLE:
+ * ====================================
+ * public doSomething(): void {
+ *   try {
+ *     executeSomething();
+ *   } catch (ex) {
+ *     this.log.exception(ex);
+ * }
+ *
+ *
+ *
+ * EXCEPTION HANDLING WITH CONTEXT - EXAMPLE:
+ * ==========================================
+ * public doSomething(): void {
+ *   try {
+ *     executeSomething();
+ *   } catch (ex) {
+ *     this.log.exception(ex, "Some additional information / context about the scenario");
+ * }
+ *
+ *
+ *
+ * EXCEPTION HANDLING WITH RE-THROW - EXAMPLE:
+ * ==========================================
+ *
+ * in case you need to do a nested / cascaded exception-handling use the RuntimeError to re-throw and
+ * provide the previous exception as cause in the new RuntimeError to retain the information
+ *
+ *  public doSomething(): void {
+ *   try {
+ *      try {
+ *       executeSomething();
+ *     } catch (ex) {
+ *       this.log.exception(
+ *          ex,
+ *         "Some additional information / context about the scenario",
+ *        );
+ *
+ *       // maybe need to do a shut-down of some component here before re-throwing
+ *
+ *        throw ExceptionHelper.getNewRuntimeErrorFromException(ex, "ExecuteSomething failed");
```

Comment:
@m-courtin There is a re-throw here, but for the third one I wanted to have a conditional handling + re-throw on the else block like

```typescript
class X {
  public async doSomething(): Promise<number> {
    try {
      await doSubTaskThatsAPartOfDoingSomething();
    } catch (ex) {
      if (ex instanceof MyErrorThatICanHandleAndRecoverFrom) {
        // An "expected exception"
        // do something here to handle the error
        // where "handling" means that the we end up recovering
        // OK instead of having to crash. Recovery means that
        // we are confident that the second sub-task is safe to proceed with
        // despite of the error that was caught here
      } else {
        // An "unexpected exception" where we want to fail immediately
        // to avoid cascading problems
        throw new RuntimeError(
          "Doing $SOMETHING failed with: ",
          ExceptionHelper.getExceptionStack(ex),
        );
      }
    }
    const result = await doSecondAndFinalSubTask();
    return result; // 42
  }
}

```

Reply:
@petermetz: Re-structured the documentation on the cascaded / conditional handling according to your suggestion